scripts/west_commands/zspdx/writer.py

```diff
@@ -102,14 +112,28 @@ def writePackageSPDX(f, pkg, spdx_version=SPDX_VERSION_2_3):
     elif len(pkg.cfg.revision) > 0:
         f.write(f"PackageVersion: {pkg.cfg.revision}\n")
 
+    if len(pkg.cfg.supplier) > 0:
+        f.write(f"PackageSupplier: Organization: {pkg.cfg.supplier}\n")
+
     for ref in pkg.cfg.externalReferences:
         if re.fullmatch(CPE23TYPE_REGEX, ref):
             f.write(f"ExternalRef: SECURITY cpe23Type {ref}\n")
         elif re.fullmatch(PURL_REGEX, ref):
-            f.write(f"ExternalRef: PACKAGE_MANAGER purl {ref}\n")
+            f.write(f"ExternalRef: PACKAGE-MANAGER purl {ref}\n")
         else:
             log.wrn(f"Unknown external reference ({ref})")
 
+    # external ref comment:
+    if len(pkg.cfg.externalReferencesComment) > 0:
+        for ln, comment in enumerate(pkg.cfg.externalReferencesComment):
+            if ln == 0: # first line
+                f.write("ExternalRefComment: <text>")
+            f.write(f"{comment}")
+            if ln == len(pkg.cfg.externalReferencesComment)-1: # last line
+                f.write("</text>\n")
+            else:
+                f.write("\n")
```

Comment:
see other comments - this is incorrect, ExternalRefComment are to be serialized next to the ExternalRef they correspond to